omod/src/main/webapp/resources/scripts/patientsearch/patientSearchWidget.js

```diff
@@ -363,6 +363,14 @@ function PatientSearchWidget(configuration){
                             }
                         });
                     }
+                    else if (column.type === 'identifierLocation') {
+                        jq.each(patient.identifiers, function (index, patientIdentifier) {
```

Comment:
(let's add the part that matched on identifier type)

var patientIdentifierType = patientIdentifier.identifierType;
// find all non-voided patientIdentifiers of whose type matches that defined in column.value
if (!patientIdentifier.voided && patientIdentifierType != null && column.value && column.value===patientIdentifierType.uuid) {